Add a dependency management checklist (#10)
tenzir/.github: contributing.md

  _how_ it addresses the underlying issue, and what _side-effects_ it might
  have.

## Dependency Management

> When introducing a new or updating an existing dependency, please consider the
> following tasks:

Reviewer: ```suggestion
When introducing a new dependency or updating an existing one, please consider the
following tasks:
```

> - **CMake:** Reason whether the dependency is a public, private, or interface

Reviewer: ```suggestion
- **CMake:** Determine whether the dependency is a public, private, or interface
```

  depdency or existing targets.

> - **Nix**: Update the Nix environment to link against the new depdendency.

Reviewer: ```suggestion
- **Nix**: Update the Nix environment to link against the proper dependency.
```

> - **Dockerfile**: Update `Dockerfile` and `Dockerfile_prebuilt` to install new
>   dependencies.

Reviewer: ```suggestion
- **Dockerfile**: Update `Dockerfile` and `Dockerfile_prebuilt` to reflect the
  dependency changes.
```

> - **CI**: Install new dependencies in GitHub Actions workflow files, including

Reviewer: ```suggestion
- **CI**: Update the dependencies in GitHub Actions workflow files, including
```

  the Jupyter Notebook workflow.

> - **Homebrew**: Notify the Homebrew package maintainer about the new
>   dependency.

Reviewer: ```suggestion
- **Homebrew**: Notify the Homebrew package maintainer about the dependency
  change.
```

> - **Debian**: Notify the Debian package maintainer about the new dependency.

Reviewer: ```suggestion
- **Debian**: Notify the Debian package maintainer about the dependency changes.
```

#### Continuous Integration

We use GitHub Actions to build and test each commit. Merging a pull request
requires that all checks pass for the latest commit in the branch. GitHub
displays the status of the individual checks in the pull request. Passing all

> checks is mandatory for every pull request.

Reviewer 2: There is a slight redundancy here.

> Merging a pull request requires that all checks pass for the latest commit in the branch
> Passing all checks is mandatory for every pull request.
